config/config.go

```diff
@@ -498,43 +486,45 @@ func (c Cluster) Config() (*Config, error) {
 		config.VPCRef = fmt.Sprintf("%q", config.VPCID)
 	}
 
+	//Assume Internet Gateway does not exist, reference by logical name
+	config.InternetGatewayRef = fmt.Sprintf(`{ "Ref" : %q }`, config.InternetGatewayLogicalName)
+	if config.InternetGatewayID != "" {
+		//This means this Internet Gateway already exists, and we can reference it directly by ID
+		config.InternetGatewayRef = fmt.Sprintf("%q", config.InternetGatewayID)
+	}
+
```

Comment:
nit: I'd like this even more if this is achieved by a function like the below so that we can eliminate the mutable field/state `InternetGatewayRef`:
```golang
func (s DeploymentSettings) InternetGatewayRef() string {
     if config.InternetGatewayId != "" {
         return fmt.Sprintf("%q", config.InternetGatewayID);
     } else {
         return fmt.Sprintf(`{ "Ref" : %q }`, config.InternetGatewayLogicalName)
     }
}
```